server/rooms.js

```diff
@@ -741,25 +741,23 @@ class GlobalRoom extends BasicRoom {
 	}
 
 	/**
-	 * @param {User} p1
-	 * @param {User} p2
+	 * @param {User[]} players
 	 * @param {GameRoom} room
 	 * @param {AnyObject} options
 	 */
-	onCreateBattleRoom(p1, p2, room, options) {
+	onCreateBattleRoom(players, room, options) {
 		if (Config.reportbattles) {
 			let reportRoom = Rooms(Config.reportbattles === true ? 'lobby' : Config.reportbattles);
 			if (reportRoom) {
+				const reportPlayers = players.map(p => p.getIdentity()).join('|');
```

Comment:
In case you missed my comment in the dev room: "room.players" upon the call of onCreateBattleRoom does not have all 2 or more players in there. I tested and figured that onCreateBattleRoom is called _before_ you do "addPlayer p2"([here](https://github.com/Zarel/Pokemon-Showdown/blob/master/server/room-battle.js#L510)) and therefore only have the 1st player in the battle log. Perhaps you can take in p1, p2 and conditionally p3 and p4 to avoid this.

Reply:
Does this matter? `players` is passed into `onCreateBattleRoom`; there should be no need to read `room.players`, right?

Reply:
Oh, I see now, it's where `players` is built.